<a href="https://colab.research.google.com/github/mks2607/Data_mks/blob/mks2607/A_B_test.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [5]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats

In [4]:
data = pd.read_csv('/content/website_wata.csv')
data

Page Views  Session Duration  Bounce Rate Traffic Source  Time on Page  \
0              5         11.051381     0.230652        Organic      3.890460   
1              4          3.429316     0.391001         Social      8.478174   
2              4          1.621052     0.397986        Organic      9.636170   
3              5          3.629279     0.180458        Organic      2.071925   
4              5          4.235843     0.291541           Paid      1.960654   
...          ...               ...          ...            ...           ...   
1995           1          2.724513     0.207187       Referral      1.324206   
1996           3          0.392856     0.095559        Organic      3.824416   
1997           4          9.899823     0.446622        Organic      1.288675   
1998           3          0.393319     0.278340           Paid      5.037584   
1999           3          0.882638     0.338026         Direct      5.186908   

      Previous Visits  Conversion Rate  
0                   3              1.0  
1                   0              1.0  
2                   2              1.0  
3                   3              1.0  
4                   5              1.0  
...               ...              ...  
1995                2              1.0  
1996                1              1.0  
1997                1              1.0  
1998                2              1.0  
1999                3              1.0  

[2000 rows x 7 columns]

1. Цель: Определить, как различные источники трафика (например, органический, платный, социальный) влияют на конверсию и время на странице.
2. Методология - по источнику трафика.
3.


---


3. Метрики: Какие метрики мы будем использовать для оценки результатов теста? Например, количество кликов, конверсии, средний чек и т.д.
4. Методология: Как мы будем разделять пользователей на группы A и B? Это может быть случайное распределение или по какому-то другому критерию.

In [7]:
# Определите ваши тестовые группы на основе источника трафика
control_group = data[data['Traffic Source'] == 'Organic']
treatment_group = data[data['Traffic Source'] == 'Paid']

In [8]:
# Замените 'Conversion Rate' на метрику, которую вы хотите проанализировать
metric = 'Conversion Rate'

In [9]:
# Предполагая, что вы хотите проверить, выше ли конверсия в тестовой группе
t_stat, p_value = stats.ttest_ind(treatment_group[metric], control_group[metric])

In [10]:
# Вывод результатов
print("T-статистика:", t_stat)
print("P-значение:", p_value)

T-статистика: -0.8354075416210155
P-значение: 0.4036529355017471


Интерпретация результатов:
T-статистика (-0.8354):

T-статистика показывает, насколько сильно различаются средние значения двух групп по сравнению с их вариацией. В данном случае, значение -0.8354 указывает на то, что среднее значение метрики в тестовой группе (например, конверсии) немного ниже, чем в контрольной группе, но не очень значительно.
P-значение (0.4037):

P-значение показывает вероятность получения наблюдаемого результата (или более экстремального), если нулевая гипотеза верна. В данном случае, нулевая гипотеза обычно утверждает, что нет разницы между группами.
Если P-значение больше уровня значимости (обычно 0.05), это означает, что у вас недостаточно доказательств, чтобы отвергнуть нулевую гипотезу. В вашем случае P-значение 0.4037 значительно больше 0.05, что указывает на то, что разница между группами не является статистически значимой.
Вывод:
На основании полученных результатов можно сделать вывод, что нет значимого различия в конверсии между контрольной группой (органический трафик) и тестовой группой (платный трафик). Это может означать, что платный трафик не приводит к более высокой конверсии по сравнению с органическим трафиком в данном тесте.